lib/task/migrate/arUpgradeSqlTask.class.php

```diff
@@ -246,6 +250,165 @@ protected function execute($arguments = [], $options = [])
+                        "content to this AtoM version's default content? (y/N)",
+                    ],
+                    'QUESTION_LARGE',
+                    false
+                )
+            )
+        ) {
+            $this->recreateMissingStaticPages($options);
+
+            $changeCount = count($this->resetOrCheckForChangedStaticPages(true));
+
+            if (empty($changeCount)) {
+                $this->logSection('upgrade-sql', "Static pages in database already match this AtoM version's.");
+            } else {
+                $this->resetOrCheckForChangedStaticPages();
+
+                $this->logSection('upgrade-sql', sprintf('Reset %d static page(s).', $changeCount));
+            }
+        }
+    }
+
+    protected function recreateMissingStaticPages($options)
+    {
+        // Ignore missing static pages when no confirmation mode is on so it has to be approved manually
+        if ($options['no-confirmation']) {
+            return;
+        }
+
+        foreach (['about', 'privacy'] as $slug) {
+            // Prompt to retore static page if its missing
+            if (null === QubitStaticPage::getBySlug($slug)
+                    && $this->askConfirmation(
+                        [
+                            sprintf('The %s page has been deleted. Do you wish to recreate it? (y/N)', $slug),
+                        ],
+                        'QUESTION_LARGE',
+                        false
+                    )
+            ) {
+                $page = new QubitStaticPage();
+                $page->slug = $slug;
+                $page->sourceCulture = 'en';
+                $page->save();
+            }
+        }
+    }
+
+    /**
+     * If "check mode" is off then, for all standard static pages, reset
+     * translations in the database to match the current AtoM default values.
+     *
+     * If "check mode" is on then, rather than resetting translations in the
+     * database, return an array noting which static pages, indicated by slug,
+     * have been changed.
+     *
+     * @param boolean $checkMode  True if to execute in "check mode"
+     *
+     * @return array  array with changed slugs as key
+     */
+    protected function resetOrCheckForChangedStaticPages($checkMode = false)
+    {
+        $changes = [];
+
+        foreach ($this->parseStaticPageFixtureData() as $slug => $translationData) {
+            $page = QubitStaticPage::getBySlug($slug);
+
+            // Only check if page still exists in the database (the privacy page can be deleted)
+            if (!isset($page)) {
+                continue;
+            }
+
+            // Check each standard translation against the database version of it
+            foreach ($translationData as $culture => $translations) {
+                // Check page title
+                if ($translations['title'] != $page->getTitle(['culture' => $culture])) {
+                    if (!$checkMode) {
```

Comment:
Ohhhh, I didn't realize there was a check mode when I did the first code review - the original two method setup (one for testing for changes, and one for doing the change) might have been fine. Sorry if I made you do unnecessary refactoring! 😬

Reply:
I think it's fine the way it is though, no need to change it back.

Reply:
It's all good... it's more elegant now!